prow/git/v2/interactor.go

```diff
@@ -222,16 +222,17 @@ func (i *interactor) BranchExists(branch string) bool {
 	return err == nil
 }
 
-func (i *interactor) CommitExists(sha string) (bool, error) {
-	i.logger.WithField("SHA", sha).Info("Checking if SHA exists")
-	_, err := i.executor.Run("branch", "--contains", sha)
-	if err != nil && strings.Contains(err.Error(), "no such commit") {
-		return false, nil
-	} else if err != nil {
-		return false, fmt.Errorf("Unable to check if commit exists: %v", err)
+func (i *interactor) ObjectExists(sha string) (bool, error) {
+	i.logger.WithField("SHA", sha).Info("Checking if Git object exists")
+	output, err := i.executor.Run("cat-file", "-e", sha)
+	// This will only error if the Git object does not exist.
+	if err != nil {
+		if string(output) != "" {
+			i.logger.WithError(err).WithField("SHA", sha).Debug(output)
```

Comment:
Don't you want this:
  if s := string(output); s != "" {
    i.logger ... Debug(s)

Reply:
Oh, right. Will update.

Reply:
Done.